### PR TITLE
[QAA-85][Detox] Refactor & improve code

### PR DESCRIPTION
**Pull request by abdurrahman-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached. (no need)
- [X] **Covered by automatic tests.** 
- [X] **Impact of the changes:** 
  - All detox tests

### 📝 Description

_Refactor Detox code and improve code:_
- rename page file and add application class 
- create application initialising function
- Various code improvments

### ❓ Context

- **JIRA or GitHub link**: 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-mobile/e2e/page/discover/discover.page.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  getElementByText,
+  openDeeplink,
+  waitForElementByText,
+} from "../../helpers";
+import { expect } from "detox";
+
+const baseLink = "discover/";
+
+export default class DiscoverPage {
+  discoverBanner = () => getElementById("discover-banner");
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/discover/discover.page.ts:12]: this element is not in use anywhere

File: apps/ledger-live-mobile/e2e/page/manager/manager.page.ts [hunk 2]
```diff
@@ -27,7 +33,11 @@ export default class ManagerPage {
   }
 
   async waitForManagerPageToLoad() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/manager/manager.page.ts:35]: this method is not in use anywhere, do we need to keep it anyway?

File: apps/ledger-live-mobile/e2e/page/manager/manager.page.ts [hunk 3]
```diff
@@ -27,7 +33,11 @@ export default class ManagerPage {
...
-    await waitFor(this.managerTitle()).toBeVisible();
+    await waitForElementById(this.managerTitleId);
+  }
+
+  async expectManagerPage() {
+    await expect(getElementById(this.managerTitleId)).toBeVisible();
   }
 
   async selectSetupNewDevice() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/manager/manager.page.ts:43]: same question

File: apps/ledger-live-mobile/e2e/page/trade/receive.page.ts [hunk 4]
```diff
@@ -38,22 +38,34 @@ export default class ReceivePage {
...
+
+  async expectSecondStep(networks: string[]) {
+    await expect(this.step2HeaderTitle()).toBeVisible();
+    await expect(this.step2Networks()).toBeVisible();
+    for (const network of networks) {
+      await expect(getElementById(this.currencyNameId(network))).toBeVisible();
+    }
+  }
+
   async selectCurrencyRow(currencyId: string) {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/trade/receive.page.ts:62]: this method is not in use anywhere, do we need to keep it anyway?

File: apps/ledger-live-mobile/e2e/page/trade/stake.page.ts [hunk 5]
```diff
@@ -8,45 +8,56 @@ export default class StakePage {
...
-  }
-
-  async selectAccount(accountId: string) {
-    const id = "account-card-" + accountId;
+    const id = this.currencyRow(currencyId);
     await waitForElementById(id);
     await tapById(id);
   }
 
   async delegationStart() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/trade/stake.page.ts:23]: this method is not in use anywhere, do we need to keep it anyway?

File: apps/ledger-live-mobile/e2e/page/trade/swap.page.ts [hunk 6]
```diff
@@ -22,49 +23,62 @@ export default class SwapFormPage {
...
 
-  navigateToSwapForm() {
-    return tapByElement(this.swapFormTab());
+  async expectSwapPage() {
+    await expect(this.swapFormTab()).toBeVisible();
   }
 
-  navigateToSwapHistory() {
-    return tapByElement(this.swapHistoryTab());
+  async navigateToSwapForm() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/trade/swap.page.ts:34]: this method is not in use anywhere, do we need to keep it anyway?

File: apps/ledger-live-mobile/e2e/page/trade/swap.page.ts [hunk 7]
```diff
@@ -22,49 +23,62 @@ export default class SwapFormPage {
...
+    await tapByElement(this.swapFormTab());
   }
 
-  openSourceAccountSelector() {
-    return tapByElement(this.swapSourceSelector());
+  async navigateToSwapHistory() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/trade/swap.page.ts:38]: same question

File: apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts [hunk 8]
```diff
@@ -33,21 +38,25 @@ export default class PortfolioPage {
...
-  openTransferMenu() {
-    return tapByElement(this.transferButton());
+  async openTransferMenu() {
+    await tapByElement(this.transferButton());
   }
 
-  navigateToSwapFromTransferMenu() {
-    // await scrollToText("Exchange crypto to crypto.", "transfer-scroll-list"); // FIXME: scroll not working
-    return tapByElement(this.swapTransferMenuButton());
+  async navigateToSwapFromTransferMenu() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts:49]: same remark

File: apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts [hunk 9]
```diff
@@ -33,21 +38,25 @@ export default class PortfolioPage {
...
+  async waitForPortfolioPageToLoad() {
+    await waitForElementById(this.portfolioSettingsButtonId, 120000);
+  }
+
+  async expectPortfolioEmpty() {
+    await expect(this.portfolioSettingsButton()).toBeVisible();
+    await expect(this.emptyPortfolioList()).toBeVisible();
   }
 
   async navigateToSendFromTransferMenu() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts:62]: same remark

File: apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts [hunk 10]
```diff
@@ -56,39 +65,39 @@ export default class PortfolioPage {
...
 
-  openMyLedger() {
-    return tapByElement(getElementById("TabBarManager"));
+  async openMyLedger() {
+    await tapById(this.managerTabBarId);
   }
 
-  openEarnApp() {
-    return tapByElement(this.earnButton());
+  async openEarnApp() {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts:99]: same remark

File: apps/ledger-live-mobile/e2e/specs/addAccounts/addAccount.spec.ts [hunk 11]
```diff
@@ -1,51 +1,41 @@
-import { expect } from "detox";
 import { knownDevice } from "../../models/devices";
-import { loadBleState, loadConfig } from "../../bridge/server";
-import PortfolioPage from "../../models/wallet/portfolioPage";
-import AccountPage from "../../models/accounts/accountPage";
-
 import DeviceAction from "../../models/DeviceAction";
-import AddAccountDrawer from "../../models/accounts/addAccountDrawer";
-import { getElementByText, waitForElementByText } from "../../helpers";
+import { Application } from "../../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/addAccounts/addAccount.spec.ts:3]: import could be shortened to "../../page"

File: apps/ledger-live-mobile/e2e/specs/deeplinks.spec.ts [hunk 12]
```diff
@@ -1,35 +1,13 @@
...
-let receivePage: ReceivePage;
-let common: Common;
-
-const ethereumLong = "Ethereum";
-const bitcoinLong = "Bitcoin";
-const arbitrumLong = "Arbitrum";
-const bobaLong = "Boba";
+import { device } from "detox";
+import { isAndroid } from "../helpers";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/deeplinks.spec.ts:3]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/delegate/cosmos.spec.ts [hunk 13]
```diff
@@ -1,76 +1,57 @@
...
-import { loadAccounts, loadBleState, loadConfig } from "../../bridge/server";
-import PortfolioPage from "../../models/wallet/portfolioPage";
-import StakePage from "../../models/trade/stakePage";
+  COSMOS_MIN_FEES,
+  COSMOS_MIN_SAFE,
+  formattedAmount,
+  getAccountUnit,
+  initTestAccounts,
+} from "../../models/currencies";
+import { Application } from "../../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/delegate/cosmos.spec.ts:9]: import could be shortened to "../../page"

File: apps/ledger-live-mobile/e2e/specs/getAllureEnvironment.spec.ts [hunk 14]
```diff
@@ -1,10 +1,11 @@
 import fs from "fs";
-import { getEnvs, getFlags, loadConfig } from "../bridge/server";
-import PortfolioPage from "../models/wallet/portfolioPage";
+import { getEnvs, getFlags } from "../bridge/server";
 import { Feature, FeatureId } from "@ledgerhq/types-live";
 import { EnvName } from "@ledgerhq/live-env";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/getAllureEnvironment.spec.ts:5]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/languageChange.spec.ts [hunk 15]
```diff
@@ -1,18 +1,12 @@
-import { expect } from "detox";
-import PortfolioPage from "../models/wallet/portfolioPage";
-import SettingsPage from "../models/settings/settingsPage";
-import GeneralSettingsPage from "../models/settings/generalSettingsPage";
-import { loadConfig } from "../bridge/server";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/languageChange.spec.ts:1]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/manager.spec.ts [hunk 16]
```diff
@@ -1,41 +1,35 @@
-import { loadBleState, loadConfig } from "../bridge/server";
-import PortfolioPage from "../models/wallet/portfolioPage";
 import DeviceAction from "../models/DeviceAction";
-import ManagerPage from "../models/manager/managerPage";
 import { knownDevice } from "../models/devices";
 import { deviceInfo155 as deviceInfo } from "@ledgerhq/live-common/apps/mock";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/manager.spec.ts:4]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/market.spec.ts [hunk 17]
```diff
@@ -1,45 +1,37 @@
-import { expect } from "detox";
-import PortfolioPage from "../models/wallet/portfolioPage";
-import MarketPage from "../models/market/marketPage";
-import GetDevicePage from "../models/discover/getDevicePage";
-import { loadConfig } from "../bridge/server";
-import { getElementByText } from "../helpers";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/market.spec.ts:1]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/nftGallery.spec.ts [hunk 18]
```diff
@@ -1,94 +1,79 @@
 import { expect } from "detox";
-import PortfolioPage from "../models/wallet/portfolioPage";
-import WalletTabNavigatorPage from "../models/wallet/walletTabNavigator";
-import NftViewerPage from "../models/nft/nftViewerPage";
-import NftGalleryPage from "../models/wallet/nftGalleryPage";
 import { tapByElement, tapByText, waitForElementById } from "../helpers";
-import ReceivePage from "../models/trade/receivePage";
-import { loadConfig } from "../bridge/server";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/nftGallery.spec.ts:3]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/onboarding.spec.ts [hunk 19]
```diff
@@ -1,17 +1,15 @@
-import { device, expect } from "detox";
-import OnboardingSteps from "../models/onboarding/onboardingSteps";
-import PortfolioPage from "../models/wallet/portfolioPage";
+import { device } from "detox";
 import { isAndroid } from "../helpers";
 import { launchApp } from "../setup";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/onboarding.spec.ts:4]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/onboardingReadOnly.spec.ts [hunk 20]
```diff
@@ -1,30 +1,22 @@
-import { expect, device } from "detox";
-import OnboardingSteps from "../models/onboarding/onboardingSteps";
-import PortfolioPage from "../models/wallet/portfolioPage";
-import MarketPage from "../models/market/marketPage";
-import GetDevicePage from "../models/discover/getDevicePage";
+import { device } from "detox";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/onboardingReadOnly.spec.ts:2]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/password.spec.ts [hunk 21]
```diff
@@ -1,52 +1,40 @@
-import { device, expect } from "detox";
-import PortfolioPage from "../models/wallet/portfolioPage";
-import SettingsPage from "../models/settings/settingsPage";
-import GeneralSettingsPage from "../models/settings/generalSettingsPage";
-import PasswordEntryPage from "../models/passwordEntryPage";
-import { loadConfig } from "../bridge/server";
+import { device } from "detox";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/password.spec.ts:2]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/receive/currencies.spec.ts [hunk 22]
```diff
@@ -1,64 +1,47 @@
-import { loadBleState, loadConfig } from "../../bridge/server";
-import PortfolioPage from "../../models/wallet/portfolioPage";
-import ReceivePage from "../../models/trade/receivePage";
 import DeviceAction from "../../models/DeviceAction";
 import { knownDevice } from "../../models/devices";
-import Common from "../../models/common";
-import { waitForElementById } from "../../helpers";
 import { getCryptoCurrencyById } from "@ledgerhq/live-common/currencies/index";
+import { Application } from "../../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/receive/currencies.spec.ts:4]: import could be shortened to "../../page"

File: apps/ledger-live-mobile/e2e/specs/receive/receiveFlow.spec.ts [hunk 23]
```diff
@@ -1,101 +1,86 @@
-import PortfolioPage from "../../models/wallet/portfolioPage";
-import ReceivePage from "../../models/trade/receivePage";
-import { loadBleState, loadConfig } from "../../bridge/server";
 import DeviceAction from "../../models/DeviceAction";
 import { knownDevice } from "../../models/devices";
-import Common from "../../models/common";
+import { Application } from "../../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/receive/receiveFlow.spec.ts:3]: import could be shortened to "../../page"

File: apps/ledger-live-mobile/e2e/specs/send/currencies.spec.ts [hunk 24]
```diff
@@ -1,28 +1,18 @@
...
 import { CryptoCurrencyId } from "@ledgerhq/types-cryptoassets";
-import Common, { formattedAmount } from "../../models/common";
-import { getDefaultAccountName } from "@ledgerhq/live-wallet/accountName";
+import {
+  formattedAmount,
+  getAccountName,
+  getAccountUnit,
+  initTestAccounts,
+} from "../../models/currencies";
+import { Application } from "../../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/send/currencies.spec.ts:11]: import could be shortened to "../../page"

File: apps/ledger-live-mobile/e2e/specs/unknown-currency-resilience.spec.ts [hunk 25]
```diff
@@ -1,47 +1,32 @@
...
 import { toAccountRaw } from "@ledgerhq/live-common/account/index";
-import { genAccount } from "@ledgerhq/live-common/mock/account";
-import {
-  getCryptoCurrencyById,
-  setSupportedCurrencies,
-} from "@ledgerhq/live-common/currencies/index";
-import { loadConfig } from "../bridge/server";
-import PortfolioPage from "../models/wallet/portfolioPage";
 import { loadAccountsRaw } from "../bridge/server";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/unknown-currency-resilience.spec.ts:3]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/specs/wallet-api.spec.ts [hunk 26]
```diff
@@ -1,35 +1,23 @@
-import { by, web } from "detox"; // this is because we need to use both the jest expect and the detox.expect version, which has some different assertions
-import { loadConfig } from "../bridge/server";
+import { by, web } from "detox";
 import { describeifAndroid } from "../helpers";
-import PortfolioPage from "../models/wallet/portfolioPage";
-import DiscoverPage from "../models/discover/discoverPage";
-import LiveAppWebview from "../models/liveApps/liveAppWebview";
-import CryptoDrawer from "../models/liveApps/cryptoDrawer";
 import { stopDummyServer } from "@ledgerhq/test-utils";
+import { Application } from "../page/index";
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-mobile/e2e/specs/wallet-api.spec.ts:4]: import could be shortened to "../page"

File: apps/ledger-live-mobile/e2e/page/accounts/accounts.page.ts [hunk 27]
```diff
@@ -2,18 +2,19 @@ import { getElementById, openDeeplink, waitForElementById, tapByElement } from "
...
+  listTitle = "accounts-list-title";
 
   async openViaDeeplink() {
     await openDeeplink(baseLink);
   }
   async waitForAccountsPageToLoad() {
-    await waitForElementById("accounts-list-title");
+    await waitForElementById(this.listTitle);
   }
   async waitForAccountsCoinPageToLoad(coin: string) {
```
[the remark this answers is not in the document]
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/accounts/accounts.page.ts:16]: It is used [here](https://github.com/LedgerHQ/ledger-live/blob/5e9090beac898823346286e35b2220bdadf3dd42/apps/ledger-live-mobile/e2e/specs/deeplinks.spec.ts#L56)

File: apps/ledger-live-mobile/e2e/page/manager/manager.page.ts (hunk 3 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/manager/manager.page.ts:43]: It's part of a path that should be tested

File: apps/ledger-live-mobile/e2e/page/trade/stake.page.ts (hunk 5 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/trade/stake.page.ts:23]: It's part of a path that should be tested. (No prior delegation in account)

File: apps/ledger-live-mobile/e2e/page/accounts/account.page.ts [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+    await waitForElementById(this.accountTitleId(assetName));
+    await waitForElementById(this.accountAssetId(assetName));
+  }
+
+  async openViaDeeplink(currencyLong?: string) {
+    const link = currencyLong ? baseLink + currencyParam + currencyLong : baseLink;
+    await openDeeplink(link);
+  }
+
+  async openAccountSettings() {
```
[the remark this answers is not in the document]
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/accounts/account.page.ts:36]: It's part of a path that should be tested

File: apps/ledger-live-mobile/e2e/page/discover/discover.page.ts (hunk 1 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/discover/discover.page.ts:12]: Fixed

File: apps/ledger-live-mobile/e2e/page/manager/manager.page.ts (hunk 2 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/manager/manager.page.ts:35]: Deleted

File: apps/ledger-live-mobile/e2e/page/trade/receive.page.ts (hunk 4 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/trade/receive.page.ts:62]: Deleted

File: apps/ledger-live-mobile/e2e/page/trade/swap.page.ts (hunk 6 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/trade/swap.page.ts:34]: It's part of a path that should be tested.

File: apps/ledger-live-mobile/e2e/page/trade/swap.page.ts (hunk 7 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/trade/swap.page.ts:38]: It's part of a path that should be tested.

File: apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts (hunk 8 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts:49]: It's part of a path that should be tested.

File: apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts (hunk 10 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/wallet/portfolio.page.ts:99]: It's part of a path that should be tested.

**Review by ypolishchuk-ledger — DISMISSED** [review]

**Review by mcoursier-ledger — DISMISSED** [review]

File: apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts [hunk 29]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  async startAccountsDiscovery() {
+    await waitForElementById(this.continueButtonId);
+  }
+
+  async expectAccountDiscovery(currency: string, index: number) {
+    const accountName = `${capitalize(currency)} ${index + 1}`;
+    await expect(this.accountCardId(this.accountId(currency, index))).toBeVisible();
+    await expect(this.accountTitleId(accountName)).toHaveText(accountName);
```
Comment (KVNLS) [inline apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts:38]: Do we need to use await here? It seems to be a synchronous operation,  isn't it?

File: apps/ledger-live-mobile/e2e/helpers.ts [hunk 30]
```diff
@@ -65,31 +68,39 @@ export async function clearTextByElement(elem: Detox.NativeElement) {
   return elem.clearText();
```
Comment (KVNLS) [inline apps/ledger-live-mobile/e2e/helpers.ts:68]: Don't we want to use await, here?
https://wix.github.io/Detox/docs/next/api/actions/#cleartext

File: apps/ledger-live-mobile/e2e/page/discover/discover.page.ts [hunk 31]
```diff
@@ -0,0 +1,26 @@
...
+  async openViaDeeplink(discoverApps = "") {
+    await openDeeplink(baseLink + discoverApps);
+  }
+
+  async expectUrl(url: string) {
+    await expect(getElementByText(url)).toBeVisible();
+  }
+
+  async expectDiscoverPage() {
+    await expect(this.discoverPageHeader()).toBeVisible();
```
Comment (KVNLS) [inline apps/ledger-live-mobile/e2e/page/discover/discover.page.ts:24]: Same here, I don't think that we need to use await, then expectDiscoverPage could be synchronous

**Review by KVNLS — CHANGES_REQUESTED** [review]
Maybe I missed something but it seems that we overuse async/await at some places. Is there a reason with Detox to do it?

File: apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts (hunk 29 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/accounts/addAccount.drawer.ts:38]: There are from Detox's expect and are asynchronous. 
- [toBeVisible](https://wix.github.io/Detox/docs/next/api/expect#tobevisible) 
- [toHaveText](https://wix.github.io/Detox/docs/next/api/expect#tohavetext)

File: apps/ledger-live-mobile/e2e/page/discover/discover.page.ts (hunk 31 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/discover/discover.page.ts:24]: Same, [toBeVisible](https://wix.github.io/Detox/docs/next/api/expect#tobevisible) from Detox's asynchronous expect.

File: apps/ledger-live-mobile/e2e/helpers.ts (hunk 30 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/helpers.ts]: Fixed

**Review by ypolishchuk-ledger — APPROVED** [review]

**Review by KVNLS — APPROVED** [review]

**Review by Justkant — APPROVED** [review]

**Pull request merged**